lib/logdna.rb

```diff
@@ -82,78 +59,68 @@ def level=(value)
       @level = value
     end
 
-    def log(msg=nil, opts={})
-      loggerExist?
-      message = msg
-      message = yield if msg.nil? && block_given?
-      @response = @@client.buffer(message, default_opts.merge(opts).merge({
-            timestamp: (Time.now.to_f * 1000).to_i
-        }))
-      'Saved'
+    def log(message, opts = {})
+      if message.empty?
+        puts "Your logline cannot be empty"
+        return
+      end
+      message = message.to_s.encode("UTF-8")
+      @client.write_to_buffer(message, default_opts.merge(opts).merge(
+                                         timestamp: (Time.now.to_f * 1000).to_i
+                                       ))
     end
 
     Resources::LOG_LEVELS.each do |lvl|
       name = lvl.downcase
 
-      define_method name do |msg=nil, opts={}, &block|
-        self.log(msg, opts.merge({
-          level: lvl,
-        }), &block)
+      define_method name do |msg = nil, opts = {}, &block|
+        log(msg, opts.merge(
+                   level: lvl
+                 ), &block)
       end
 
       define_method "#{name}?" do
-        return Resources::LOG_LEVELS[self.level] == lvl if self.level.is_a? Numeric
-        self.level == lvl
+        return Resources::LOG_LEVELS[level] == lvl if level.is_a? Numeric
+
+        assign_level == lvl
```

Comment:
This shows where your rename doesn't make sense; `assign_level` is a method, this line doesn't make sense. This should be checking the level var.